### PR TITLE
Export layer notification methods in v8

### PR DESCRIPTION
**Pull request by GeoffCoxMSFT**
To better support removing deep imports in ODSP.

## Changes
- added top-level export of Layer notification methods
- added top-level export of Layer getStyles as getLayerStyles

## Issues
Updates #24456

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/f3104008db3243375c8e3ab0c152adacafab46d0'>f3104008db3243375c8e3ab0c152adacafab46d0</a></sub>
<div id="bundleSizeComment9423"/>

**Review by dzearing — APPROVED** [review]

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>BaseButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/BaseButton-mount_base.html">976</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/BaseButton-mount.html">939</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Breadcrumb</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/Breadcrumb-mount_base.html">2895</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/Breadcrumb-mount.html">2890</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Checkbox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/Checkbox-mount_base.html">2667</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/Checkbox-mount.html">2634</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>CheckboxBase</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/CheckboxBase-mount_base.html">2385</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/CheckboxBase-mount.html">2336</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ChoiceGroup</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/ChoiceGroup-mount_base.html">4828</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/ChoiceGroup-mount.html">4877</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ComboBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/ComboBox-mount_base.html">1072</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/ComboBox-mount.html">973</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>CommandBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/CommandBar-mount_base.html">10813</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/CommandBar-mount.html">10881</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>ContextualMenu</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/ContextualMenu-mount_base.html">12685</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/ContextualMenu-mount.html">12926</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DefaultButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/DefaultButton-mount_base.html">1175</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/DefaultButton-mount.html">1165</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRow</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/DetailsRow-mount_base.html">3891</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/DetailsRow-mount.html">3846</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowFast</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/DetailsRowFast-mount_base.html">3947</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/DetailsRowFast-mount.html">3899</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowNoStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/DetailsRowNoStyles-mount_base.html">3699</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/DetailsRowNoStyles-mount.html">3772</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Dialog</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/Dialog-mount_base.html">3206</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/Dialog-mount.html">3148</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DocumentCardTitle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/DocumentCardTitle-mount_base.html">187</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/DocumentCardTitle-mount.html">176</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Dropdown</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/Dropdown-mount_base.html">3436</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/Dropdown-mount.html">3440</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusTrapZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/FocusTrapZone-mount_base.html">1941</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/FocusTrapZone-mount.html">1822</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/FocusZone-mount_base.html">1862</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/FocusZone-mount.html">1785</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>IconButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/IconButton-mount_base.html">1772</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/IconButton-mount.html">1775</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Label</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/Label-mount_base.html">351</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/Label-mount.html">374</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Layer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/Layer-mount_base.html">4556</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/Layer-mount.html">4664</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Link</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/Link-mount_base.html">496</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/Link-mount.html">509</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MenuButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/MenuButton-mount_base.html">1509</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/MenuButton-mount.html">1562</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MessageBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/MessageBar-mount_base.html">2303</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/MessageBar-mount.html">2187</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Nav</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/Nav-mount_base.html">3505</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/Nav-mount.html">3504</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>OverflowSet</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/OverflowSet-mount_base.html">1113</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/OverflowSet-mount.html">1131</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Panel</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/Panel-mount_base.html">2564</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/Panel-mount.html">2575</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/Persona-mount_base.html">991</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/Persona-mount.html">1065</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Pivot</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/Pivot-mount_base.html">1463</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/Pivot-mount.html">1470</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>PrimaryButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/PrimaryButton-mount_base.html">1327</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/PrimaryButton-mount.html">1373</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Rating</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/Rating-mount_base.html">7953</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/Rating-mount.html">7863</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SearchBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/SearchBox-mount_base.html">1274</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/SearchBox-mount.html">1327</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Shimmer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/Shimmer-mount_base.html">2873</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/Shimmer-mount.html">2850</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Slider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/Slider-mount_base.html">1970</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/Slider-mount.html">2015</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/SpinButton-mount_base.html">5088</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/SpinButton-mount.html">5527</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Spinner</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/Spinner-mount_base.html">448</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/Spinner-mount.html">437</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SplitButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/SplitButton-mount_base.html">3288</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/SplitButton-mount.html">3170</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Stack</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/Stack-mount_base.html">507</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/Stack-mount.html">540</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithIntrinsicChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/StackWithIntrinsicChildren-mount_base.html">2303</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/StackWithIntrinsicChildren-mount.html">2304</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithTextChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/StackWithTextChildren-mount_base.html">5375</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/StackWithTextChildren-mount.html">5229</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SwatchColorPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/SwatchColorPicker-mount_base.html">12218</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/SwatchColorPicker-mount.html">11884</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TagPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/TagPicker-mount_base.html">2714</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/TagPicker-mount.html">2631</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TeachingBubble</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/TeachingBubble-mount_base.html">101777</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/TeachingBubble-mount.html">101292</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Text</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/Text-mount_base.html">435</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/Text-mount.html">434</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TextField</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/TextField-mount_base.html">1365</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/TextField-mount.html">1385</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/ThemeProvider-mount_base.html">1287</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/ThemeProvider-mount.html">1340</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/ThemeProvider-virtual-rerender_base.html">771</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/ThemeProvider-virtual-rerender.html">795</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/ThemeProvider-virtual-rerender-with-unmount_base.html">2082</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/ThemeProvider-virtual-rerender-with-unmount.html">2115</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Toggle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/Toggle-mount_base.html">813</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/Toggle-mount.html">793</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>buttonNative</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/buttonNative-mount_base.html">126</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24485/perf-test/buttonNative-mount.html">140</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9423"/>

File: packages/react/src/components/Layer/Layer.notification.ts [hunk 1]
```diff
@@ -8,8 +8,8 @@ let _defaultHostSelector: string | undefined = `#${defaultHostId}`;
 
 /**
  * Register a layer for a given host id
- * @param hostId Id of the layer host
- * @param layer Layer instance
+ * @param hostId - Id of the layer host
+ * @param layer - Layer instance
```
Comment (spmonahan) [inline packages/react/src/components/Layer/Layer.notification.ts:12]: Param name is incorrect.

File: packages/react/src/components/Layer/Layer.notification.ts [hunk 2]
```diff
@@ -29,8 +29,8 @@ export function registerLayer(hostId: string, callback: () => void) {
 
 /**
  * Unregister a layer for a given host id
- * @param hostId Id of the layer host
- * @param layer Layer instance
+ * @param hostId - Id of the layer host
+ * @param layer - Layer instance
```
Comment (spmonahan) [inline packages/react/src/components/Layer/Layer.notification.ts:33]: Param name is incorrect.

**Review by spmonahan — APPROVED** [review]

**Pull request merged**